### PR TITLE
Linking inverses

### LINKED ISSUE #35: Parse "INVERSE" attributes from IFC schema
Many (previously existing) functionalities can only be implemented, when the INVERSE attributes in the IFC schema are parsed and corresponding members generated. One of such functionalities is #34 .

### PR DESCRIPTION
**Pull request by pjanck**
Superseeding #274 and #197 - both went the wrong direction and did not bear fruit. So I backtracked a bit to a commit that made sense and continued there.

Fixes #35 .

Notes:
- unit tests for the generator are broken - will provide new issues for that.
- EXPRESS lib still needs refactoring, as well as IFC geometry converters.
- the linking of inverses currently cannot be done in parallel (`pragma omp for`) - will make an improvement suggestion

### CODE REVIEW

File: Core/src/IfcGeometryConverter/FaceConverter.h [hunk 1]
```diff
@@ -1239,8 +1239,50 @@ namespace OpenInfraPlatform {
...
+
+							if (face.template isOfType<typename IfcEntityTypesT::IfcIndexedPolygonalFaceWithVoids>())
+								BLUE_LOG(warning) << "Ignoring voids at " << face->getErrorLog();
+						}
+
+						itemData->open_or_closed_polyhedrons.push_back(polygon);
+						return;
+					}
+
+					throw oip::UnhandledException(tessItem);
```
Comment (hechth) [inline Core/src/IfcGeometryConverter/FaceConverter.h:1285]: This should better be in a seperate branch/PR, but only for the future. Try to keep things separated.

File: CurrentlyExcluded/OPEN_INFRA_PLATFORM/Core/DataManagement/Export/ExportIfc4x1ExcelReport.cpp [hunk 2]
```diff
@@ -32,7 +32,6 @@
 #include "IfcGeometryConverter/UnitConverter.h"
 #include "IfcGeometryConverter/IfcPeekStepReader.h"
 
-#include "EXPRESS\EXPRESSAttribute.h"
 #include "EXPRESS\EXPRESSEntity.h"
```
Comment (hechth) [inline CurrentlyExcluded/OPEN_INFRA_PLATFORM/Core/DataManagement/Export/ExportIfc4x1ExcelReport.cpp:35]: This is currently excluded code!

File: UnitTests/Generator/graph/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#    TUM Open Infra Platform is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+include(CreateUnitTests)
```
Comment (hechth) [inline UnitTests/Generator/graph/CMakeLists.txt:18]: What is the connection to the unit tests here?

**Review by hechth — COMMENTED** [review]
Good but contains a lot of not directly related content.

File: UnitTests/Generator/graph/CMakeLists.txt (hunk 3 above)
Comment (pjanck) [reply UnitTests/Generator/graph/CMakeLists.txt:18]: I've provided a unit test for the new class `Graph`.

File: CurrentlyExcluded/OPEN_INFRA_PLATFORM/Core/DataManagement/Export/ExportIfc4x1ExcelReport.cpp (hunk 2 above)
Comment (pjanck) [reply CurrentlyExcluded/OPEN_INFRA_PLATFORM/Core/DataManagement/Export/ExportIfc4x1ExcelReport.cpp:35]: Agreed.

File: Core/src/IfcGeometryConverter/FaceConverter.h (hunk 1 above)
Comment (pjanck) [reply Core/src/IfcGeometryConverter/FaceConverter.h:1285]: Agreed. It was a hot fix for one of my projects and I did not feel like switsching branches and waiting for the compile ...

**Pull request merged**